### PR TITLE
chore: additional logging in alerts and reports

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

This adds more logging for alerts and reports, to better understand downstream exceptions. 

Of particular note is that our logs show this messaging often:
image
 
So we wanted to mark that task as having failed. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by pkdotson — APPROVED** [review]
LGTM!

File: superset/utils/core.py [hunk 1]
```diff
@@ -1020,6 +1021,7 @@ def send_mime_email(
     if smtp_user and smtp_password:
         smtp.login(smtp_user, smtp_password)
     logger.debug("Sent an email to %s", str(e_to))
+    logger.info("Sent email notification content is %s", mime_msg.as_string())
```
Comment (dpgaspar) [inline superset/utils/core.py:1024]: don't think we should log the mime_msg itself

File: superset/tasks/scheduler.py [hunk 2]
```diff
@@ -85,10 +85,12 @@ def execute(report_schedule_id: int, scheduled_dttm: str) -> None:
         logger.exception(
             "An unexpected occurred while executing the report: %s", task_id
         )
+        execute.update_state(state="FAILURE")
```
Comment (dpgaspar) [inline superset/tasks/scheduler.py:88]: Can you add a test to assert this?

File: superset/tasks/scheduler.py [hunk 3]
```diff
@@ -85,10 +85,12 @@ def execute(report_schedule_id: int, scheduled_dttm: str) -> None:
...
     except CommandException:
         logger.exception(
             "A downstream exception occurred while generating" " a report: %s", task_id
         )
+        execute.update_state(state="FAILURE")
```
Comment (dpgaspar) [inline superset/tasks/scheduler.py:93]: can you add a test to assert this?

File: superset/reports/notifications/email.py [hunk 4]
```diff
@@ -207,6 +207,8 @@ def send(self) -> None:
                 dryrun=False,
                 header_data=content.header_data,
             )
-            logger.info("Report sent to email")
+            logger.info(
+                "Report sent to email, notification content is %s", content.header_data
```
[the remark this answers is not in the document]
Comment (AAfghahi) [reply superset/reports/notifications/email.py:211]: We log it in other places, so I think that it is.

File: superset/utils/core.py (hunk 1 above)
Comment (AAfghahi) [reply superset/utils/core.py:1024]: ok, I think this one we can actually delete. Ty

**Review by eschutho — APPROVED** [review]

**Pull request merged**